packetbeat/protos/protos.go

```diff
@@ -231,11 +231,17 @@ func (protocols ProtocolsStruct) BpfFilter(with_vlans bool, with_icmp bool) stri
 
 func (protos ProtocolsStruct) Register(proto Protocol, plugin ProtocolPlugin) {
 	protos.all[proto] = plugin
+	var success bool = false
```

Comment:
@andrewkroh seems plugin like memcache implement both tcp and udp protocol, so i think it is not a good idea to use type switch here,what you think?